pkg/experiment/query_backend/metrics.go

```diff
@@ -0,0 +1,22 @@
+package query_backend
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type metrics struct {
+	mismatchingTenantDataset *prometheus.CounterVec
+}
+
+func newMetrics(reg prometheus.Registerer) *metrics {
+	m := &metrics{
+		mismatchingTenantDataset: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "pyroscope",
+				Subsystem: "query_backend",
+				Name:      "invalid_tenant_query_total",
+			}, []string{"tenant"}),
```

Comment:
I think we probably don't need the break down per tenant in the metric, it basically should not happen for any tenant, if it does we have to identify via logs anyhow what exactly happened.